crates/biome_yaml_parser/src/lexer/mod.rs

```diff
@@ -98,28 +100,42 @@ impl<'src> YamlLexer<'src> {
             self.current_byte()
                 .is_some_and(|c| self.is_first_plain_char(c, context))
         );
+        let start = self.position();
         self.advance_char_unchecked();
+        let mut num_utf8_char = 1;
         while let Some(c) = self.current_byte() {
             // https://yaml.org/spec/1.2.2/#rule-ns-plain-char
             if is_plain_safe(c, context) && c != b':' && c != b'#' {
                 self.advance_char_unchecked();
+                num_utf8_char += 1;
             } else if is_non_space_char(c) && self.peek_byte().is_some_and(|c| c == b'#') {
                 self.advance_char_unchecked();
                 self.advance(1); // '#'
+                num_utf8_char += 2;
```

Comment:
Do we have a better way to count utf8 chars?

Reply:
Not really. But I wouldn't recommend this approach anyway. A better way to deal with this would be to do `self.position() - start`

Reply:
Yeah I think this can be better address with the analyzer. It will cover not just a plain node but flow (JSON like nodes) as well

Reply:
It should though, I could've sworn that self.position() returned the byte position

Reply:
Ah right, my bad, I should have been more clear. `p.position()` returns byte position, but I'm not aware of any easy way to convert that byte distance to Unicode char distance. Like, if the byte distance is 1024, but the lexer has no confidence in the number of Unicode char in this range, so the distance in Unicode char might be smaller than 1024.

On the other hand, as Ema advised, I think this one might be better fitted in an analyzer rule. Let's hold this discussion until then